runtime/src/main/java/org/corfudb/runtime/view/RuntimeLayout.java

```diff
@@ -109,8 +110,9 @@ private IClient getClient(final Class<? extends IClient> clientClass,
             endpointClientMap.computeIfAbsent(endpoint, s -> {
                 try {
                     Constructor<? extends IClient> ctor =
-                            clientClass.getDeclaredConstructor(IClientRouter.class, long.class);
-                    IClient inst = ctor.newInstance(getRuntime().getRouter(endpoint), layout.getEpoch());
+                            clientClass.getDeclaredConstructor(IClientRouter.class, long.class, UUID.class);
```

Comment:
Just pass the whole layout instead of passing the epoch and cluster id separately ?

Reply:
I feel like for router it is a sufficient enough information